substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/substitute/UnsafeAutomaticSubstitutionProcessor.java

```diff
@@ -682,6 +702,26 @@ private static ResolvedJavaField extractValueStoreField(ValueNode valueNode, Kin
             if (signExtendFieldStoreUsages.count() == 1) {
                 offsetField = signExtendFieldStoreUsages.first().field();
             }
+        } else {
+            // Determine if the usage of the offset escapes from local usage.
+            for (Node valueNodeUsage : valueNodeUsages) {
+                // Probably other valid node types could be added.
+                if (valueNodeUsage instanceof FrameState || valueNodeUsage instanceof MethodCallTargetNode) {
```

Comment:
I don't think `MethodCallTarget` is a valid reason without further qualification. We need to have a whitelist of allowed target methods here too, like `Unsafe` methods that read and write memory using the offset.

Reply:
Okie dokie, I can make that a little more constrained, for sure.